src/flake8/options/config.py

```diff
@@ -120,6 +121,16 @@ def parse_config(
         LOG.debug('Option "%s" returned value: %r', option_name, value)
 
         final_value = option.normalize(value, cfg_dir)
+
+        if option_name in {"ignore", "extend-ignore"}:
+            for error_code in final_value:
+                if not VALID_CODE.match(error_code):
+                    raise configparser.ParsingError(
```

Comment:
I don't think this is the right type -- the others raise `ValueError` so we should probably do similarly?

Reply:
e.g. supplying an undecodable config file throws a ParsingError (```Raised when a configuration file does not follow legal syntax.```), but I agree that in this context it makes more sense to raise a ValueError since it'll also be raised if a user supplies an invalid code in otherwise valid syntax.